src/renderer/components/hooks/useCheckboxes.js

```diff
@@ -0,0 +1,106 @@
+import { checkValues } from '../TriStateCheckbox/TriStateCheckbox';
+import { useState } from 'react';
+
+const makeCheckboxesStateObj = (extCloud) => {
+  return (extCloud?.namespaces || []).reduce((acc, namespace) => {
+    acc[namespace.name] = false;
+    return acc;
+  }, {});
+};
+export const makeCheckboxesInitialState = (extCloud) => {
+  return {
+    parent: false,
+    children: makeCheckboxesStateObj(extCloud),
+  };
+};
+
+/**
+ * @param {boolean} initialState.parent
+ * @param {Object} initialState.children {[key]: boolean, ....} pairs. Mainly it's result of makeCheckboxesStateObj
+ */
+export function useCheckboxes(initialState) {
+  const [checkboxesState, setCheckboxesState] = useState(initialState);
+
+  // set parent checkbox state based on changes of children checkboxes
+  const setParentCheckboxState = (children) => {
+    const childrenCheckboxes = Object.values(children);
+
+    if (
+      childrenCheckboxes.every((el) => el === false) ||
+      childrenCheckboxes.every((el) => el === true)
+    ) {
+      return childrenCheckboxes[0];
+    }
+    if (childrenCheckboxes.some((el) => el === false)) {
+      return true;
+    }
+    return checkboxesState.parent;
+  };
+
+  const getNewChildren = (parentCheckedStatus) => {
+    const newChildren = { ...checkboxesState.children };
+
+    Object.keys(checkboxesState.children).forEach((name) => {
+      newChildren[name] = parentCheckedStatus;
+    });
+    return newChildren;
+  };
+
+  const getParentCheckboxValue = () => {
+    const childrenCheckboxes = Object.values(checkboxesState.children);
+
+    if (
+      checkboxesState.parent &&
+      childrenCheckboxes.some((el) => el === false) &&
+      childrenCheckboxes.some((el) => el === true)
+    ) {
+      return checkValues.MIXED;
+    }
+    if (checkboxesState.parent) {
+      return checkValues.CHECKED;
+    } else {
+      return checkValues.UNCHECKED;
+    }
```

Comment:
I think this might be a lint error?

```suggestion
    if (checkboxesState.parent) {
      return checkValues.CHECKED;
    }

    return checkValues.UNCHECKED;
```